zebra-chain/src/orchard/shielded_data.rs

```diff
@@ -38,6 +38,28 @@ pub struct ShieldedData {
     pub binding_sig: Signature<Binding>,
 }
 
+impl ShieldedData {
+    /// Iterate over the [`Action`]s for the [`AuthorizedAction`]s in this transaction.
+    pub fn actions(&self) -> impl Iterator<Item = &Action> {
+        self.actions.actions()
+    }
+
+    /// Collect the [`Nullifier`]s for this transaction, if it contains
+    /// [`Action`]s.
+    pub fn nullifiers(&self) -> impl Iterator<Item = &Nullifier> {
+        self.actions().map(|action| &action.nullifier)
+    }
+}
+
+impl AtLeastOne<AuthorizedAction> {
+    /// Iterate over the [`Action`]s of the `AuthorizedAction` for this transaction,
+    /// returning them as their generic type.
```

Comment:
```suggestion
    /// Iterate over the [`Action`] of each [`AuthorizedAction`].
```

Just a suggestion, not sure if it's worth changing. My reasoning was that `AtLeastOne<AuthorizedAction>` is not exactly a transaction, so it made sense to be a bit more general in the description.

Reply:
We should definitely make this change. There is no generic in any `Action` type, that's a copy-paste from `sapling::Spend<AnchorV: AnchorVariant>`

Reply:
Thanks, i think @teor2345 made this one as well in the other PR but it seems i missed it.